js/web/lib/wasm/jsep/webgpu/ops/3rd-party/conv_backprop_webgpu.ts

```diff
@@ -0,0 +1,380 @@
+        `@group(0) @binding(1) var<storage, read> W: array<${isVec4 ? 'vec4<f32>' : 'f32'}>;`
+      ];
+      let declareFunctions = `
+  fn setOutputAtIndex(flatIndex : u32, value : ${isVec4 ? 'vec4<f32>' : 'f32'}) {
+    result[flatIndex] = ${isVec4 ? 'vec4<f32>' : 'f32'}(value);
+  }`;
+      if (hasBias) {
+        declareInputs.push(`@group(0) @binding(2) var<storage, read> bias: array<${isVec4 ? 'vec4<f32>' : 'f32'}>;`);
+        declareFunctions += `
+    fn getBiasByOutputCoords(coords : vec4<u32>) -> ${isVec4 ? 'vec4<f32>' : 'f32'} {
+      return bias[coords.${isChannelsLast ? 'w' : 'y'}${isVec4 ? '/ 4' : ''}];
+    }`;
+      }
+      const wIndicesHelper = createIndicesHelper('W', inputs[1].dims);
+      const dyIndicesHelper = createIndicesHelper('Dy', inputs[0].dims);
+      const outputIndicesHelper = createIndicesHelper('output', outputShape);
+      const codeSnippet4 = `{
+        let batch: u32 = global_id.z / outShape[1];
+        let r = global_id.z % outShape[1];
+        let c = global_id.y * ${workPerThread};
+        let d1: u32 = global_id.x * 4;
+
+        let dyCorner = vec2<i32>(i32(r), i32(c)) - vec2<i32>(pads);
+
+        // Convolve dy(?, ?, d2) with w(:, :, d1, d2) to compute dx(xR, xC, d1).
+        // ? = to be determined. : = across all values in that axis.
+        var dotProd: array<vec4<f32>, ${workPerThread}>;
+        for (var i = 0; i < ${workPerThread}; i++) {
+          dotProd[i] = vec4<f32>(0.0);
+        }
+        for (var wR: u32 = 0; wR < filterDims[0]; wR = wR + 1) {
+          var dyR = f32(dyCorner.x + wR) / f32(strides.x);
+          let wRPerm: u32= filterDims[0] - 1 - wR;
+          if (dyR < 0.0 || dyR >= f32(outBackprop[1]) ||
+              fract(dyR) > 0.0) {
+            continue;
+          }
+          let idyR: u32 = u32(dyR);
+
+          for (var wC: u32 = 0; wC < filterDims[1]; wC = wC + 1) {
+            let dyC = f32(dyCorner.y + wC) / f32(strides.y);
+            let dyC2 = f32(dyCorner.y + 1 + wC) / f32(strides.y);
+            let wCPerm: u32 = filterDims[1] - 1 - wC;
+            var bDyCVal = true;
+            var bDyCVal2 = true;
+            if (dyC < 0.0 || dyC >= f32(outBackprop[2]) ||
+                fract(dyC) > 0.0) {
+              bDyCVal = false;
+            }
+            if (dyC2 < 0.0 || dyC2 >= f32(outBackprop[2]) ||
+                fract(dyC2) > 0.0) {
+              bDyCVal2 = false;
+            }
+
+            let idyC: u32 = u32(dyC);
+            let idyC2: u32 = u32(dyC2);
+            if (bDyCVal && bDyCVal2) {
+              let d2Length = outBackprop[3];
+              for (var d2 :u32 = 0; d2 < d2Length; d2 = d2 + 4) {
+                ${
+          wIndicesHelper.indicesVariableDeclaration(
+              'wIndices0',
+              isChannelsLast ? ['d2', 'wRPerm', 'wCPerm', 'd1'] :
+                               [
+                                 'd2', 'd1', 'wRPerm', 'wCPerm'
+                               ])};
```

Comment:
I assume that `isChannelsLast` is for X (input[0]), not for W (input[1]). And for W it should always be channel last , because you always transpose it in conv-transpose.ts. if you don't tranpose, it should be always align with spec https://onnx.ai/onnx/operators/onnx__ConvTranspose.html (C x M/group x kH x kW)